src/main/java/org/broadinstitute/consent/http/service/DarCollectionService.java

```diff
@@ -335,6 +322,20 @@ public List<DarCollectionSummary> getSummariesForRoleName(User user, String user
     return summaries;
   }
 
+  private List<Integer> getDatasetIdsForUserAndRole(User user, UserRoles role) {
+    List<Integer> roleDacIds = user.getRoles().stream()
+        .filter(ur -> ur.getRoleId().equals(role.getRoleId()))
+        .map(UserRole::getDacId)
+        .filter(Objects::nonNull)
+        .toList();
+    return Stream.of(roleDacIds)
+        .filter(arr -> !arr.isEmpty())
```

Comment:
Whoops, guess I got the logic backward in my version. Up to you which way is clearer but using `Predicate.not()` should work here too
```suggestion
        .filter(Predicate.not(List::isEmpty))
```

Reply:
I like that, will give it a shot